File: library/src/main/java/com/yzyfdf/library/activity/SampleContract.java
package com.yzyfdf.library.activity;


import com.yzyfdf.library.base.BaseModel;
import com.yzyfdf.library.base.BasePresenter;
import com.yzyfdf.library.base.BaseView;
import com.yzyfdf.library.rx.BaseRespose;

import io.reactivex.Observable;

public interface SampleContract {

    interface Model extends BaseModel {
        Observable<BaseRespose> get(String s);
    }

    interface View extends BaseView {
        void returnS(BaseRespose bean);
    }

    abstract static class Presenter extends BasePresenter<View, Model> {
        public abstract void get(String s);
    }
}

File: library/src/main/java/com/yzyfdf/library/sample/SampleContract.kt
package com.yzyfdf.library.sample


import com.yzyfdf.library.base.BaseModel
import com.yzyfdf.library.base.BasePresenter
import com.yzyfdf.library.base.BaseView
import com.yzyfdf.library.rx.BaseRespose

import io.reactivex.Observable

interface SampleContract {

    interface Model : BaseModel {
        operator fun get(s: String): Observable<BaseRespose<*>>
    }

    interface View : BaseView {
        fun returnS(bean: BaseRespose<*>)
    }

    abstract class Presenter : BasePresenter<View, Model>() {
        abstract operator fun get(s: String)
    }
}
